### ISSUE TITLE
Downgrade cargo package when latest version is yanked

### ISSUE THREAD

**Issue by alex** [dependabot/dependabot-core#2155]
In a Rust project, I merged a PR from dependabot to upgrade `bitflags` to 1.0.5. Later, the 1.0.5 release was yanked. I think in this case it'd be appropriate for dependabot to _lower_ the version to 1.0.4, the otherwise latest version. I think this behavior makes sense because if I run `cargo update` locally, I get a downgrade. You can see this with the following repository: https://github.com/alex/csv-sql

**Comment by snarfed** [comment]
Yes! Beyond Rust, this would be useful for all package ecosystems that support yanking. Eg I recently got bit by a hang bug in grpc's pip package, https://github.com/grpc/grpc/issues/31885. They eventually yanked the affected release version, 1.52.0, but dependabot had already upgraded my projects, so I had to debug myself and eventually figure out the affected dependency and what to do about it, https://github.com/grpc/grpc/issues/31885#issuecomment-1461244306. Would definitely have been nice if dependabot spared me that!

**Comment by alex** [comment]
I don't believe this has become stale.
